DNN Platform/DotNetNuke.Abstractions/Logging/IEventLogService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Logging
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The event log service provides APIs for
+    /// managing existing Event Logs in the database.
+    /// </summary>
+    public interface IEventLogService
+    {
+        /// <summary>
+        /// Clears the Event Log.
+        /// </summary>
+        void ClearLog();
+
+        /// <summary>
+        /// Deletes a specific Event Log.
+        /// </summary>
+        /// <param name="logInfo">the log info.</param>
+        void DeleteLog(ILogInfo logInfo);
+
+        /// <summary>
+        /// Get the Event Log.
+        /// </summary>
+        /// <param name="portalID">The portal id.</param>
+        /// <param name="logType">The log type.</param>
+        /// <param name="pageSize">The page size.</param>
+        /// <param name="pageIndex">The page index.</param>
+        /// <param name="totalRecords">The total records.</param>
+        /// <returns>The logs.</returns>
+        IEnumerable<ILogInfo> GetLogs(int portalID, string logType, int pageSize, int pageIndex, ref int totalRecords);
+
+        /// <summary>
+        /// Get a single log.
+        /// </summary>
+        /// <param name="log">The log to retrieve.</param>
+        /// <param name="returnType">The return type.</param>
+        /// <returns>The single log.</returns>
+        object GetSingleLog(ILogInfo log, LoggingProviderReturnType returnType);
```

Comment:
Sorry I've missed this in previous reviews.  I know this effort is lift-and-shift, but I'm going to have a hard time letting this one pass…

I'd lean towards getting rid of the `LoggerProviderReturnType` altogether, and just always returning `ILogInfo`.  We can then add a `ToXmlNode` method to `ILogInfo`.

How does that feel to y'all, @ahoefling @mitchelsellers?

Reply:
I am always happy to remove stuff that doesn't make sense anymore. While working through these APIs this is one that caught my attention and I have all the questions on why we still need

Reply:
The XML return type is in use by the Event Log Persona Bar component, when it's creating the attachment to email log events:
https://github.com/dnnsoftware/Dnn.Platform/blob/80b6b2c56c06a058a8995196f8fa8375a28c9508/Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/AdminLogs/AdminLogsController.cs#L256-L270

The implementation always fills a `LogInfo` object, and then optionally converts it to an `XmlNode` by calling `LogInfo.Serialize()` and feeding it to `XmlDocument.LoadXml`.

Also, rather than taking a whole `ILogInfo`, we can really just take a `Guid` (or `string`, since that's how we represent the GUID on `ILogInfo`), since that's all that's used to retrieve the data.

Reply:
Thank you for the explanation, is there any action I can take on this right now or are we waiting for @mitchelsellers to respond?

Reply:
I would say you're free to tackle it if you have time now.

Reply:
I just pushed changes to address this concern. Let's open up new threads to comment on the code